### PR TITLE
21585: Refactored client and types, MAJOR

### PR DESCRIPTION
**Pull request by fulpm**
- Types are now auto generated via the Howso Engine instead of using openapi
- Trainee methods are now generated via the Howso Engine
- Refactored type and client packages structure
- The WasmClient is now named HowsoWorkerClient

### CODE REVIEW

File: src/types/schemas/ReactAggregate.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+ *
+ * ReactAggregate
+ *
+ * Computes, caches, and returns specified details and feature prediction statistics such as mean decrease in accuracy (mda), residuals (accuracy, mean absolute error),
+ *  precision, recall, etc. returns details and feature prediction stats for all features in the format of feature -> assoc stat -> value
+ */
+import type { ReactAggregateDetails } from "./ReactAggregateDetails";
+
+export type ReactAggregateRequest = {
+  /*
```
Comment (lancegliser) [inline src/types/schemas/ReactAggregate.ts:12]: I'd love to see these property comments use the full `/**`. JS docs and IDE interpreters would pass them along.

File: src/types/shims/FeatureOriginalType.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
+export interface BooleanType {
```
Comment (lancegliser) [inline src/types/shims/FeatureOriginalType.ts:1]: I'm a little worried about potential name collision or confusion on these with our core types. We probably need to prefix all of these with `OriginalType`. So `OriginalTypeBoolean`, or the reverse.

File: src/types/shims/React.ts
Comment (lancegliser) [inline src/types/shims/React.ts:1]: I wonder if we need to make a point of running these sorts of long objects (all objects?) through alpha sorting.

File: src/types/trainee.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+export interface TraineeIdentity {
```
Comment (lancegliser) [inline src/types/trainee.ts:1]: Let's preference `type` for these.

**Review by lancegliser — CHANGES_REQUESTED** [review]

File: src/client/worker/HowsoWorkerClient.ts [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+    let num_trained = 0;
+    let status = null;
+    const ablated_indices: number[] = [];
+
+    // Batch scale the requests
+    await batcher(
+      async function* (this: HowsoWorkerClient, size: number) {
+        let offset = 0;
+        while (offset < cases.length) {
+          const { payload: response } = await this.train(trainee.id, {
```
Comment (lancegliser) [inline src/client/worker/HowsoWorkerClient.ts:565]: No access to the warning portion of this response. With `trace: true`, I did see it. Still, I wonder if we can't do better. Going to see about adding a `Logger` option that can auto log warnings if supplied internally.

File: src/types/shims/React.ts
Comment (fulpm) [reply src/types/shims/React.ts:1]: Will come with the removal of the shims, going to leave as is for now

**Review by lancegliser — APPROVED** [review]

**Pull request merged**